docs/developer-notes.md

```diff
@@ -139,6 +139,14 @@ CATcher uses the OAuth 2.0 protocol to authenticate users. Below is a summary of
 
 The authentication process is kicked off in the `AuthComponent`, but the code that co-ordinates steps 1 and 2 can be found in [`oauth.ts`](../oauth.ts)(For Electron) or `AuthService`(For Web). Step 2 requires a client secret granted to CATcher. To protect this, we run a web service, [gatekeeper](https://github.com/CATcher-org/gatekeeper) that executes step 2 on behalf of the client CATcher app.
 
+# Handling Releases
+
```

Comment:
Before explaining how to use the GitHub Actions' workflow to automate the release,
I think we should explicitly list the tasks that need to be done for a release.

This will be helpful as a reference checklist, especially if a manual release needs to be done (for some reason).

```
Releasing a new version of CATcher involves the following steps:
- Update CATcher's version number in `package.json`
- Build the CATcher desktop application's executables for Linux, MacOS and Windows
- Build the CATcher web app, and then deploy it onto our GitHub Pages site (https://catcher-org.github.io/CATcher/)
- Create a release on GitHub, upload the desktop app's executables and write a
  changelog that describes any new features or bug fixes.
```